### DISCUSSION TITLE
Multiple steps reaction

### DISCUSSION THREAD

**Discussion by AmirrezaHBSH** [firemodels/fds#12828] [General]
Is it possible to write multiple steps reaction into one single step reaction? For example propane reacts with O2 gives CO and H2O and C (soot) then soot and CO react with O2 again and produces CO2. how we can write these three steps reactions into one single step reaction? Note that these reactions are fast reactions

**Comment by mcgratta** [comment]
You just sum the left and right hand sides of the 3 reactions.

**Reply by AmirrezaHBSH** [reply]
Thank you for your response, however, I have a gas phase consists of CO, CO2, CH4, H2 with different compositions 60 %wt, 29% wt, 9% wt, 2% wt (on mass basis) respectively. Each of these species undergoes a specified reaction therefore there is multiple step reactions. Now by considering this information if just sum the left and right hand side, it means that I neglect the composition of components. 
Could you please help me with this regard?

**Reply by drjfloyd** [reply]
If the CO,CO2,CH4,H2 mixture  is initially present or being injected with the same composition, then you can just define a lumped species for that mixture. Since all the species are just C,H,O, and N you can then just have a single simple chemistry REAC input that uses that lumped species.

**Reply by AmirrezaHBSH** [reply]
Actually CO, CO2, CH4, H2 are the products of pyrolysis reaction (Char+gas). The reactions for each component are: 

CH_4+2O_2→CO_2+2H_2 O

CO+0.5O_2→CO_2

CO_2→CO+0.5O_2

H_2+0.5O_2→2H_2 O

By using these reactions and the composition of each spices, 60 %wt, 29% wt, 9% wt, 2% wt (on mass basis) respectively I am trying to write single step reaction. Could you please help me with these information?

**Reply by drjfloyd** [reply]
It isn;t clear exactly what you are trying to model:

If you are just intializing the domain with some mixture of gasses and not predicting pyrolysis then:

Say you have a mixture that is 50 % H2 and 50 % CO2 and the domain intially has a 10 % mass fraction of that mixture. You could simply have as inputs:

&SPEC ID='HYDROGEN',LUMPED_COMPONENT_ONLY=T/
&SPEC ID='CARBON DIOXIDE',LUMPED_COMPONENT_ONLY=T/
&SPEC ID='MY MIXTURE',SPEC_ID='HYDROGEN','CARBON DIOXIDE',VOLUME_FRACTION=0.5,0.5/

&REAC ID='MY REAC',FUEL='MY MIXTURE',CO_YIELD=...../

If you are predicting each of those pyrolysis reactions, then you will need to define each reaction independently on a MATL line. If you are looking to define a single pyrolysis reaction that represents the production of the four species, that would be a research effort for you to figure out how to make the math work.
